library/swift/src/Counter.swift

```diff
@@ -0,0 +1,22 @@
+@_implementationOnly import EnvoyEngine
+import Foundation
+
+@objc
+public class Counter: NSObject {
+  private let series: String
+  private weak var engine: EnvoyEngine?
+
+  internal init(elements: [Element], engine: EnvoyEngine) {
+    self.series = elements.map{ $0.description }.joined(separator: ".")
+    self.engine = engine
+    super.init()
+  }
+
+  func increment(count: Int) {
+    guard let engine = self.engine else {
+      return
+    }
+
+    engine.recordCounter(self.series, count: numericCast(count))
```

Comment:
Why are we exposing an `Int` publicly but requiring an unsigned integer internally? Shouldn't these be the same (in which case we could avoid a cast)?

Reply:
My impression is that Swift convention strongly favors treating base numeric types as signed wherever possible. (And in Java/Kotlin, there's basically no alternative.) On the other hand, Envoy is explicit about counters accepting unsigned values only. `numericCast` imposes strong runtime checks on invalid (in this case, negative) ranges. And this keeps the two platform interfaces aligned.